torch/_inductor/quantized_lowerings.py

```diff
@@ -95,5 +102,54 @@ def _mul_epilogue(buf: torch.Tensor) -> Any:
             "_weight_int8pack_mm", choices, [mat1, mat2, scale], aten_layout
         )
 
+    @register_lowering(aten._weight_int4pack_mm_for_cpu, type_promotion_kind=None)  # type: ignore[misc]
+    def int4pack_mm_cpu(
+        input: torch.Tensor,
+        weight: torch.Tensor,
+        qGroupSize: int,
+        qScaleAndZeros: torch.Tensor,
+        *,
+        layout: Any = None,
+    ) -> Any:
+        _, _, _, layout, mat1, mat2 = mm_args(
+            input, weight, layout=layout, use_4x2_dim=True, mat2_transposed=True
+        )
+        assert (
+            mat1.get_dtype() in [torch.bfloat16, torch.float16, torch.float]
+            and mat2.get_dtype() == torch.uint8
+        )
+        group_size = V.graph.add_tensor_constant(
+            torch.tensor(qGroupSize, dtype=torch.int64), name=None
+        )
+        aten_layout = layout
+
+        # options to tune from
+        choices = (
+            [
+                aten__weight_int4pack_mm_cpu.bind(
+                    (mat1, mat2, group_size, qScaleAndZeros), aten_layout
+                )
+            ]
+            if use_aten_gemm_kernels()
+            else []
+        )
+
+        if (
+            len(choices) == 0
+            and inductor_config.autotune_fallback_to_aten
+            and not use_aten_gemm_kernels()
+        ):
+            log.warning("No choices for GEMM, using ATen backend as fallback")
+            return aten__weight_int4pack_mm_cpu.bind(
+                (mat1, mat2, group_size, qScaleAndZeros), aten_layout
+            ).output_node()
+
+        return autotune_select_algorithm(
+            "_weight_int4pack_mm_for_cpu",
+            choices,
+            [mat1, mat2, group_size, qScaleAndZeros],
+            aten_layout,
+        )
+
```

Comment:
How about postponing this change to your int4 WoQ GEMM template PR? Thanks!

Reply:
The lowering pass is a prerequisite to enable max-autotune. And this is the major change of this PR.

Reply:
Thanks for your response! I meant since auto-tuning for int4 WoQ GEMM would not work with the current main branch, would you delay the landing of this PR until it would?

Reply:
This PR is the prerequisite for auto-tuning of woq int4. I need to land this first before adding the auto-tuning path. Thanks.

Reply:
> This PR is the prerequisite for auto-tuning of woq int4. I need to land this first before adding the auto-tuning path. Thanks.

To be accurate, this PR actually adds the auto-tuning path in the lowering function but we have only one option, which is the aten kernel. I think it 's ok to put this part in this PR or the next PR. So, let's just keep it in this PR. Thanks.